strawberryfields/apps/graph/max_clique.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Maximum clique
+==============
+
+**Module name:** :mod:`strawberryfields.apps.graph.max_clique`
+
+.. currentmodule:: strawberryfields.apps.graph.max_clique
+
+This module provides tools for users to identify large cliques in graphs. A clique is a subgraph
+where all nodes are connected to each other. The maximum clique problem is to identify the
+largest clique in a graph. The problem is NP-Hard in a worst-case setting, which is why it is
+valuable to develop better algorithms to tackle it.
+
+The core of this module is a local search algorithm that proceeds as follows:
+
+#. A small clique in the graph is identified. The initial clique can be a single node, or it can
+   be obtained by shrinking a random subgraph, for example obtained from GBS.
+#. The clique is grown as much as possible by adding nodes connected to all nodes in the clique.
+#. When no more growth is possible, a node in the clique is swapped with a node outside of it.
+
+Steps 2-3 are repeated until a pre-determined number of steps have been completed, or until no
+more growth or swaps are possible.
+
+Summary
+-------
+
+.. autosummary::
+    local_search
+
+Code details
+^^^^^^^^^^^^
+"""
+import networkx as nx
+
+from strawberryfields.apps.graph import resize
+
+
+def local_search(
+    clique: list, graph: nx.Graph, iterations: int = 100, node_select: str = "uniform"
+) -> list:
+    """Use local search algorithm to identify large cliques.
+
+    This function implements a version of the local search algorithm given in
+    :ref:`pullan2006dynamic` and :ref:`pullan2006phased`. It proceeds by iteratively applying
+    phases of greedy growth and plateau search to the input clique subgraph. Growth is achieved
+    using the :func:`~strawberryfields.apps.graph.resize.clique_grow` function, which repeatedly
+    evaluates the set :math:`C_0` of nodes in the remainder of the graph that are connected to
+    all nodes in the clique, and selects one candidate node from :math:`C_0` to make a larger
+    clique. Plateau search is performed with the
+    :func:`~strawberryfields.apps.graph.resize.clique_swap` function, which evaluates the set
+    :math:`C_1` of nodes in the remainder of the graph that are connected to all but one of the
+    nodes in the clique, and selects one candidate from :math:`C_1` and its corresponding
+    unconnected node in the clique to swap.

```

Comment:
Slightly awkward phrasing. How about `selects one candidate from :math:`C_1` and swaps it with its corresponding node in the clique`

Reply:
Thanks, fixed!